### PR TITLE
[vectorized](pipeline) support assert num rows operator

### PR DESCRIPTION
**Pull request by zhangstar333**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/vassert_num_rows_node.cpp [hunk 1]
```diff
@@ -39,20 +37,20 @@ VAssertNumRowsNode::VAssertNumRowsNode(ObjectPool* pool, const TPlanNode& tnode,
...
+    RETURN_IF_ERROR(alloc_resource(state));
     // ISSUE-3435
     RETURN_IF_ERROR(child(0)->open(state));
     return Status::OK();
 }
 
-Status VAssertNumRowsNode::get_next(RuntimeState* state, Block* block, bool* eos) {
-    INIT_AND_SCOPE_GET_NEXT_SPAN(state->get_tracer(), _get_next_span,
-                                 "VAssertNumRowsNode::get_next");
+Status VAssertNumRowsNode::alloc_resource(RuntimeState* state) {
```
Comment (Gabriel39) [inline be/src/vec/exec/vassert_num_rows_node.cpp:46]: Seems this is useless. Can we use ExecNode::alloc_resource ?

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.08 seconds
 load time: 487 seconds
 storage size: 17123356283 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221208060022_clickbench_pr_59936.html

**Review by Gabriel39 — APPROVED** [review]
LGTM

File: be/src/pipeline/exec/assert_num_rows_operator.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+class AssertNumRowsOperatorBuilder final : public OperatorBuilder<vectorized::VAssertNumRowsNode> {
+public:
+    AssertNumRowsOperatorBuilder(int32_t id, ExecNode* node)
+            : OperatorBuilder(id, "AssertNumRowsOperatorBuilder", node) {};
+
+    OperatorPtr build_operator() override;
+};
+
+class AssertNumRowsOperator final : public Operator<AssertNumRowsOperatorBuilder> {
```
Comment (Gabriel39) [inline be/src/pipeline/exec/assert_num_rows_operator.h:35]:
```suggestion
class AssertNumRowsOperator final : public StreamingOperator<AssertNumRowsOperatorBuilder> {
```

**Review by Gabriel39 — CHANGES_REQUESTED** [review]

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**